sdk/core/core-https/src/interfaces.ts

```diff
@@ -1,20 +1,167 @@
+   */
+  interface Blob {}
+
+  interface File extends Blob {}
+
+  /** Provides a way to easily construct a set of key/value pairs representing form fields and their values, which can then be easily sent using the XMLHttpRequest.send() method. It uses the same format a form would use if the encoding type were set to "multipart/form-data". */
+  interface FormData {
+    append(name: string, value: string | Blob, fileName?: string): void;
+    delete(name: string): void;
+    get(name: string): string | File | null;
+    getAll(name: string): Array<string | File>;
+    has(name: string): boolean;
+    set(name: string, value: string | Blob, fileName?: string): void;
+    forEach(
+      callbackfn: (value: string | File, key: string, parent: FormData) => void,
+      thisArg?: any
+    ): void;
+  }
+}
+
+/**
+ * A HttpHeaders collection represented as a simple JSON object.
+ */
+export type RawHttpHeaders = { [headerName: string]: string };
+
+/**
+ * Represents a set of HTTP headers on a request/response.
+ * Header names are treated as case insensitive.
+ */
+export interface HttpHeaders extends Iterable<[string, string]> {
+  /**
+   * Returns the value of a specific header or undefined if not set.
+   * @param name The name of the header to retrieve.
+   */
+  get(name: string): string | undefined;
+  /**
+   * Returns true if the specified header exists.
+   * @param name The name of the header to check.
+   */
+  has(name: string): boolean;
+  /**
+   * Sets a specific header with a given value.
+   * @param name The name of the header to set.
+   * @param value The value to use for the header.
+   */
+  set(name: string, value: string | number): void;
+  /**
+   * Removes a specific header from the collection.
+   * @param name The name of the header to delete.
+   */
+  delete(name: string): void;
+  /**
+   * Duplicates this collection.
+   */
+  clone(): HttpHeaders;
+  /**
+   * Accesses a raw JS object that acts as a simple map
+   * of header names to values.
+   */
+  toJSON(): RawHttpHeaders;
+}
+
+/**
+ * Types of bodies supported on the request.
+ * NodeJS.ReadableStream is Node only.
+ * Blob is browser only.
+ */
+export type RequestBodyType =
+  | NodeJS.ReadableStream
+  | Blob
+  | ArrayBuffer
+  | ArrayBufferView
+  | FormData
+  | string
+  | null;
+
 /**
  * Metadata about a request being made by the pipeline.
- * TODO: this interface isn't complete, it is just a placeholder.
  */
 export interface PipelineRequest {
   /**
    * The URL to make the request to.
    */
   url: string;
+
+  /**
+   * The HTTP method to use when making the request.
+   */
+  method: HttpMethods;
+
+  /**
+   * The HTTP headers to use when making the request.
+   */
+  headers: HttpHeaders;
+
+  /**
+   * The number of milliseconds a request can take before automatically being terminated.
+   * If the request is terminated, an `AbortError` is thrown.
+   * Defaults to 0, which disables the timeout.
+   */
+  timeout: number;
```

Comment:
The way this works today, it would be each retry would use this timeout to make the request. E.g. if you set it to 2s and the retry policy tried 3 times, you would wait for 6s before totally failing.

If this is expected/desirable, I can update the comment.

/cc @bterlson

Reply:
I was wrong about this -- retry policy doesn't retry abort errors today, which is how timeout is modeled as.